vnext/ReactWindowsCore/WinRTWebSocketResource.cpp

```diff
@@ -237,15 +273,36 @@ fire_and_forget WinRTWebSocketResource::PerformWrite()
       self->m_writer.WriteBytes(arr);
     }
 
-    co_await self->m_writer.StoreAsync();
+    auto async = self->m_writer.StoreAsync();
 
-    if (self->m_writeHandler)
+    co_await lessthrow_await_adapter<DataWriterStoreOperation>{async};
+
+    if (SUCCEEDED(async.ErrorCode()))
+    {
+      if (self->m_writeHandler)
+      {
+        self->m_writeHandler(length);
+      }
+    }
+    else
+    {
+      if (self->m_errorHandler)
+      {
+        self->m_errorHandler({ GetRestrictedErrorMessage(), ErrorType::Send });
+      }
+    }
+  }
+  catch (std::exception const& e)
+  {
+    // Utf8ToUtf16 may throw
+    if (self->m_errorHandler)
     {
-      self->m_writeHandler(length);
+      self->m_errorHandler({ e.what(), ErrorType::Ping });
     }
   }
   catch (hresult_error const& e)
   {
+    //TODO: Remove after fixing unit tests exceptions.
```

Comment:
fix before checkin?

Reply:
Added to the pre-merge checklist (see description).